waspc/data/Generator/templates/sdk/wasp/server/auth/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { type ProviderId } from '../../auth/utils.js'
+import { type PrismaClient } from '@prisma/client'
+
+export type AuthHookFn =
+  | OnUserSignUpHookFn
+  | OnUserLoginHookFn
+  | OnUserLogoutHookFn
+
+type OnUserLoginHookFn = (context: {
+  providerId: ProviderId
+  prisma: PrismaClient
+}) => Promise<void> | void
+type OnUserSignUpHookFn = (context: {
+  providerId: ProviderId
+  prisma: PrismaClient
+}) => Promise<void> | void
+type OnUserLogoutHookFn = (context: {
+  providerId: ProviderId
+  prisma: PrismaClient
+}) => Promise<void> | void
+
+type HookType = 'onUserSignUp' | 'onUserLogin' | 'onUserLogout'
+
+export function defineHook<HT extends HookType>(
```

Comment:
What is the purpose of this function? Looks to me like it just returns the hook it takes, and it even stays the same type.